docs/man/rpm-scriptlets.7.scd

```diff
@@ -0,0 +1,360 @@
+Package file triggers execute once per each triggering package.
+
+*%filetriggerin*
+	Executed just after the installation of each triggering package.
+
+*%filetriggerun*
+	Executed just before the removal of each triggering package.
+
+*%filetriggerpostun*
+	Executed just after the removal of each triggering package.
+
+## Transaction File Triggers
+Transaction file triggers only execute once per transaction, regardless
+of how many packages triggered it.
+
+*%transfiletriggerin*
+	Executed once at the end of a transaction, for all matching installed
+	files (ie. from the transaction or previously installed).
+
+*%transfiletriggerun*
+	Executed once at the start of a transaction, for all matching removed
+	files.
+
+*%transfiletriggerpostun*
+	Executed once at the end of a transaction if there were matching
+	removed files.
+
+	Note: the list of triggering files is not available in this slot.
+
+# EXECUTION ORDER
+The relative execution order of scriptlets within a transaction for a
+single package upgrade is as follows. _new_ refers to a newly installed
+package version, _old_ refers to the already installed package version
+to be replaced by _new_, and _other_ means installed packages, and the
+other packages in the current transaction. (XXX FIXME: not quite right)
+
+. *%pretrans* of _new_
+. *%preuntrans* of _old_
+. *%transfiletriggerun* of _other_, set off by removal of _old_
+. *%sysusers* of _new_ (implicit)
+. *%triggerprein* of _other_, set off by install of _new_
+. *%triggerprein* of _new_, set off by _other_
+. *%pre* of _new_
+. (unpack _new_ files)
+. *%filetriggerin* of _other_ (high priority), set off by _new_
+. *%filetriggerin* of _new_ (high priority), set off by _other_
+. *%post* of _new_
+. *%triggerin* of _other_, set off by _new_
+. *%triggerin* of _new_, set off by _other_
+. *%filetriggerin* of _other_ (low priority), set off by _new_
+. *%filetriggerin* of _new_ (low priority), set off by _other_
+. *%triggerun* of _old_, set off by _other_
+. *%triggerun* of _other_, set off by _old_
+. *%filetriggerun* of _old_ (high priority), set off by _other_
+. *%filetriggerun* of _other_ (high priority), set off by _old_
+. *%preun* of _old_
+. *%filetriggerun* of _old_ (low priority), set off by _other_
+. *%filetriggerun* of _other_ (low priority), set off by _old_
+. (erase _old_ files)
+. *%filetriggerpostun* of _old_ (high priority), set off by _other_
+. *%filetriggerpostun* of _other_ (high priority), set off by _old_
+. *%postun* of _old_
+. *%filetriggerpostun* of _old_ (low priority), set off by _other_
+. *%filetriggerpostun* of _other_ (low priority), set off by _old_
+. *%posttrans* of _new_
+. *%postuntrans* of _old_
+. *%transfiletriggerpostun* of _other_, set off by _old_
+. *%transfiletriggerin* of _other_, set off by _new_
+
+# ENVIRONMENT
+*RPM_INSTALL_PREFIX*_N_
+	The _N_'th The install prefix in relocatable packages.
+
+*RPM_INSTALL_PREFIX*
+	The install prefix form in legacy relocatable packages.
```

Comment:
"form" seems like a leftover...?

Reply:
Also, I'm a bit confused by this env var. It looks like it contains a Lua table in `-p lua` scriptlets, and the *last* prefix otherwise? So maybe we should clarify that here, too...

Reply:
"form" is kinda intentional, maybe it should be "format" because the way relocations are handled was changed somewhere 20+ years ago :smile: 

For Lua scripts, there's indeed a related global table by the same name, but that's not to be confused with the environment variable. The "modern" relocation prefixes environment variables are numbered to support multiple prefixes per package, something like
```
RPM_INSTALL_PREFIX0=/aaa
RPM_INSTALL_PREFIX1=/bbb
```

In Lua that can be handled by a table, but traditional shell doesn't have arrays...

Reply:
Argh, right, how could an *environment* variable be a Lua specific variable... Dunno what I was thinking here :sweat_smile:

Reply:
...and there's another thing that wants an example, but I didn't have one on hand...

Reply:
Oh sure :+1: